### PR TITLE
Some Minor Cleanups in o.e.snapshots

### PR DESCRIPTION
**Pull request by original-brownbear**
* Some minor obvious cleanups that became possible now tht we're on JDK11
* Removing redundant `Updates` internal class

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/snapshots/SnapshotShardFailure.java [hunk 1]
```diff
@@ -155,24 +155,14 @@ private static SnapshotShardFailure constructSnapshotShardFailure(Object[] args)
 
         ShardId shardId = new ShardId(index, indexUuid != null ? indexUuid : IndexMetaData.INDEX_UUID_NA_VALUE, intShardId);
 
-        // Workaround for https://github.com/elastic/elasticsearch/issues/25878
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/snapshots/SnapshotShardFailure.java]: this was for BwC with 5.x, it's irrelevant in 8.0 now

File: server/src/main/java/org/elasticsearch/snapshots/RestoreService.java [hunk 2]
```diff
@@ -595,15 +595,15 @@ public RestoreInfo getRestoreInfo() {
     }
 
     public static class RestoreInProgressUpdater extends RoutingChangesObserver.AbstractRoutingChangesObserver {
-        private final Map<String, Updates> shardChanges = new HashMap<>();
+        private final Map<String, Map<ShardId, ShardRestoreStatus>> shardChanges = new HashMap<>();
```
Comment (andrershov) [inline server/src/main/java/org/elasticsearch/snapshots/RestoreService.java]: In general, I don't like nested maps. Here it's ok, but I'd like to have a comment on what is inside. Especially for such generic type as String.

**Review by andrershov — APPROVED** [review]
I've left one comment, otherwise LGTM.

**Comment by original-brownbear** [comment]
Thanks @andrershov !

**Pull request merged**